src/Shared/FileMatcher.cs

```diff
@@ -2603,5 +2690,26 @@ private static bool DirectoryEndsWithPattern(string directoryPath, string patter
         };
 
         internal static bool IsRecursiveDirectoryMatch(string path) => path.TrimTrailingSlashes() == recursiveDirectoryMatch;
+
+        public struct GetFilesImplementationResults
```

Comment:
This might be a good opportunity to use a C# 9 [`record`](https://docs.microsoft.com/dotnet/csharp/language-reference/builtin-types/record), IMO. Though some of the immutability stuff isn't obvious to me from a functional + perf perspective.